pkg/api/admin/operatorflags.go

```diff
@@ -0,0 +1,54 @@
+package admin
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+type operatorFlagsMergeStrategyStruct struct {
+	OperatorFlagsMergeStrategy string
+	Oc                         api.OpenShiftCluster
+}
+
+const operatorFlagsMergeStrategyDefault string = "merge"
+
+// When a cluster is edited via the PATCH Cluster Geneva Action (aka an Admin Update)
+// the flags given are treated according to the provided Update Strategy,
+// provided in operatorFlagsMergeStrategy
+
+// merge (default): The provided cluster flags are laid on top of the cluster’s existing flags.
+// reset: The provided cluster flags are laid on top of the default cluster flags,
+// essentially ‘resetting’ the flags if no new flags are provided.
+func OperatorFlagsMergeStrategy(oc *api.OpenShiftCluster, body []byte) error {
+	payload := operatorFlagsMergeStrategyStruct{
+		OperatorFlagsMergeStrategy: operatorFlagsMergeStrategyDefault,
+		Oc:                         api.OpenShiftCluster{},
+	}
+
+	err := json.Unmarshal(body, &payload)
+	if err != nil {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidRequestContent, "", "The request content was invalid and could not be deserialized: %q.", err)
+	}
+
+	// return error if OperatorFlagsMergeStrategy is not merge or reset, default is merge
+	if payload.OperatorFlagsMergeStrategy != "merge" && payload.OperatorFlagsMergeStrategy != "reset" {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "", "invalid operatorFlagsMergeStrategy '%s', can only be 'merge' or 'reset'", payload.OperatorFlagsMergeStrategy)
+	}
+	// return nil, if OperatorFlagsMergeStrategy is merge and payload has not operatorFlags
+	// return operatorFlags of payload, if OperatorFlagsMergeStrategy is merge and payload has operatorFlags
+	// return defaultOperatorFlags, if OperatorFlagsMergeStrategy is reset and payload has not operatorFlags
+	// return defaultOperatorFlags + operatorFlags of payload, if OperatorFlagsMergeStrategy is reset and payload has operatorFlags
```

Comment:
Before accessing payload.Oc.Properties.OperatorFlags, it may be a good idea to ensure that payload.Oc.Properties is not nil to prevent a potential panic ?
```
if payload.OpenShiftCluster.Properties == nil {
    return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidRequestContent, "", "The request content was invalid: Properties is nil.")
}
```

Reply:
We should skip on an empty payload, I added an extra check for this, thank you,